packages/core/src/languages/solidity/storage.ts

```diff
@@ -266,7 +273,64 @@ const encodeVariable = (
       throw new Error('large strings (>31 bytes) not supported')
     }
   } else if (variableType.encoding === 'mapping') {
-    throw new Error('mapping types not yet supported')
+    let slots = []
+    for (const [key, value] of Object.entries(variable)) {
+      // default pack type for value types
+      let type = variableType.key.split('_')[1]
+      // default key encoding for value types
+      let encodedKey: string | Uint8Array = encodeVariable(
+        key,
+        storageObj,
+        storageTypes,
+        nestedSlotOffset + parseInt(storageObj.slot as any, 10),
+        undefined,
+        variableType.key
+      )[0].val
+
+      if (variableType.key.startsWith('t_uint')) {
+        // all uints must be packed with type uint256
+        type = 'uint256'
+      } else if (variableType.key.startsWith('t_int')) {
+        // all ints must be packed with type int256
+        type = 'int256'
+      } else if (variableType.key.startsWith('t_string')) {
+        // strings do not need to be encoded
+        // pack type can be pulled from input type
+        encodedKey = key
+      } else if (variableType.key.startsWith('t_bytes')) {
+        // bytes do not need to be encoded, but must be converted from the input string
+        // pack type can be pulled straight from input type
+        encodedKey = ethers.utils.toUtf8Bytes(key)
```

Comment:
I think we actually want this line to be:
```
encodedKey = fromHexString(key)
```
This is because we want the original `key` to be of the form `0x...` (see my other comments)